Helper/Discount.php

```diff
@@ -400,4 +424,92 @@ public function getUnirgyGiftCertBalanceByCode($giftcertCode)
 
         return (float) $result;
     }
+
+    /**
+     * Check whether the Mirasvit Store Credit module is allowed for quote
+     *
+     * @param $quote
+     *
+     * @return bool
+     */
+    public function isMirasvitStoreCreditAllowed($quote)
+    {
+        if (!$this->isMirasvitStoreCreditAvailable()) {
+            return false;
+        }
+
+        return $quote->getCreditAmountUsed() > 0 && $this->getMirasvitStoreCreditBalanceAmount($quote) > 0;
+    }
+
+    /**
+     * @param      $quote
+     *
+     * @param bool $paymentOnly
+     *
+     * @return float
+     */
+    public function getMirasvitStoreCreditAmount($quote, $paymentOnly = false)
+    {
+        $miravitBalanceAmount = $this->getMirasvitStoreCreditBalanceAmount($quote);
+        if(!$paymentOnly){
+            /** @var \Mirasvit\Credit\Api\Config\CalculationConfigInterface $miravitConfig */
+            $miravitConfig = $this->mirasvitStoreCreditConfig->getInstance();
+            if ($miravitConfig->isTaxIncluded() || $miravitConfig->IsShippingIncluded()){
+                return $miravitBalanceAmount;
+            }
+        }
+
+        $toUse = $quote->getGrandTotal() + $quote->getCreditAmountUsed();
```

Comment:
@johnmccombs1 
It's unclear to me here what is going on.  I imagine that here you are looking into all the other non-Miravit store credit balances that are used, `$quote->getCreditAmountUsed()`, and subtracting that from the grand total, (i.e. I would expect that value to be negative).

So, `$toUse` could be labeled something like `$outstandingBalance`, or `$unresolvedTotal` and ultimately, you return min($unresolvedTotal, $miravitBalanceAmount).

Is this interpretation correct?